### DISCUSSION TITLE
Many requests in client service-account-roles UI using keycloak.v2 theme

### DISCUSSION THREAD

**Discussion by davidpiendl** [keycloak/keycloak#14465] [Q&A] [answered]
Hi everybody,

we currently did an update to Keycloak 19.0.1-legacy and tested the new AdminUI. There we faced an issue, which does not occure using the deprecated theme. We have multiple clients in our realm, which have different service-account-roles. If I go to one clients service-account-roles tab I face a huge amount of requests (>100) loading all available roles. Even if I want to assign a role in the modal, which opens when you click "Assign role", there are dozens of requests which sometimes trigger the rate limiting of our nginx and we run into a 429 response. We already increased the rate limiting, but depending of the UI operations of the user, it will happen from time to time.

So is this amount of requests expected, or is it a bug? Does anybody else observe a huge amount of requests in the service-account-roles management? Is this maybe related to the legacy version?

Thank you in advance and best regards
David

**Comment by ssilvert** [comment, marked as answer]
@davidpiendl Can you try out the latest nightly build?  We made some changes to fix this kind of issue but I don't know if we addressed that particular one.

https://github.com/keycloak/keycloak/releases/tag/nightly

If there is still an issue in the latest build, please open an issue at https://github.com/keycloak/keycloak-ui/issues
